### ISSUE TITLE
DateTimeRangePicker with analog clock has weird behaviour

### ISSUE THREAD

**Issue by JessArschoot** [mui/mui-x#15740]
### Search keywords

DateTimeRangePicker

### Latest version

- [X] I have tested the latest version

### Affected products

Date and Time Pickers

### Steps to reproduce

Steps:
1. Open this link to live example:[ (required)](https://codesandbox.io/embed/rxtf3j?module=/src/Demo.tsx&fontsize=12)
2. Select a Start date
3. Change time, first hours and then the seconds, when sliding the seconds, popover is closed and end date is selected

### Current behavior

The analog clock has weird behaviour, when sliding in the minutes, it suddenly selects a value and close the popover

### Expected behavior

The expected behaviour should be, on the left part only date selection. So we should be able to select immediately the date range.
In the analog clock we should only update the time of the selected date. Initially this should get a default value. When a valid time is selected we don't want to move to the other date. This is occurring the buggy behaviour I think. 

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID or Support key 💳 (optional)

89609

**Comment by michelengelen** [comment]
Thanks for opening this @JessArschoot ... it definitely is a bug. For some reason clicking inside the clock immediately selects a value and tries to switch to the next view.
It also happens on this example in the docs: [Choose time view renderer](https://mui.com/x/react-date-pickers/date-time-picker/#choose-time-view-renderer)

I'll add it to the board! 👍🏼

**Comment by JessArschoot** [comment]
I have a follow-up question concerning this DateTimeRangePicker. When I select a start date, a default time is also selected, which is what I want. However, unlike the DateRangePicker, I cannot immediately select a date range. I'm obligated to always select a time. Is this the wanted behaviour?

**Comment by LukasTy** [comment]
> I cannot immediately select a date range. I'm obligated to always select a time. Is this the wanted behaviour?

Yes, that is the baked-in view selection behavior: `start date` -> `start time` -> `end date` -> `end time`.
Do you need to have `start date` -> `end date` -> `start time` -> `end time` behavior? 🤔

**Comment by JessArschoot** [comment]
Yes that is how our users would work the most efficient.
Our application provides a "default" start and end time. For instance office hours. Those always remain the same. Only in some conditions will our users change those times. So the dates are the most important part of this DateTimeRangePicker.
Is this something we can achieve?

These are possible behaviors from our users:
- Default start and end time are ok: `start date` -> `end date`
- Start time is ok, only end time needs to be adjusted: `start date` -> `end date` -> `end time`
- End time is good, start time not: `start date` -> `start time` -> `end date`
- Everything is custom: `start date` -> `start time` -> `end date` -> `end time`

**Comment by JessArschoot** [comment]
Basically in short, we would like the opportunity to always select a date range in the DateTimeRangePicker, without having to select a time first.

**Comment by StijnG-EBO** [comment]
Adding into the discussion on efficiently using the date/time range we still observe that a lot of users are having issues with having to always follow the fixed flow start_date, start_time, end_date & end_time and they would like (as suggested by @JessArschoot ) skip the start_time & end_time and have start_date -> end_date while leaving the start_time & end_time optional.
